### PR TITLE
DT-1901: new endpoint to get full client details

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+  "clientDetails": {
+    "scope": [
+      "read"
+    ],
+    "client_id": "another-test-client-3",
+    "client_secret": "$2a$10$lBwbziQlLfiCnn8Kj1PfMujEcLdsJYlYSNJvBRO638gCYTS9yN0xm",
```
Comment (petergphillips) [inline src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json:7]: don't like this - should be excluded from extract

Comment (SimonMitchellMOJ) [reply src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json:7]: on it now!

File: src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      "client_credentials"
+    ],
+    "autoapprove": [
+      "read"
+    ],
+    "access_token_validity": 1200
+  },
+  "duplicates": [
+    {
+      "id": "another-test-client-2",
```
Comment (petergphillips) [inline src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json:18]: dubious whether we should be exposing anything other than the `id` here, can't see that @mattops would need it

Comment (SimonMitchellMOJ) [reply src/test/resources/uk/gov/justice/digital/hmpps/oauth2server/resource/api/auth_client_details.json:18]: i will take it back to just the id

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/DuplicateClientControllerIntTest.kt [hunk 3]
```diff
@@ -2,12 +2,13 @@ package uk.gov.justice.digital.hmpps.oauth2server.resource.api
...
 import org.json.JSONObject
 import org.junit.jupiter.api.Test
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
 import org.springframework.http.HttpStatus
 import org.springframework.http.MediaType
 import org.springframework.test.web.reactive.server.expectBody
 import uk.gov.justice.digital.hmpps.oauth2server.resource.IntegrationTest
 import java.util.Base64
-
+@AutoConfigureWebTestClient(timeout = "136000")
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/DuplicateClientControllerIntTest.kt:11]: accidentally left in?

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**